src/main/java/com/iexec/sms/secret/web2/Web2SecretService.java

```diff
@@ -65,14 +78,14 @@ public Optional<Web2Secret> getSecret(String ownerAddress, String secretAddress,
      *                                      with same {@code ownerAddress}/{@code secretAddress} couple already exists
      */
     public Web2Secret addSecret(String ownerAddress, String secretAddress, String secretValue) throws SecretAlreadyExistsException {
-        final Optional<Web2Secret> oSecret = getSecret(ownerAddress, secretAddress);
-        if (oSecret.isPresent()) {
+        final boolean isSecretPresent = isSecretPresent(ownerAddress, secretAddress);
+        if (isSecretPresent) {
```

Comment:
```suggestion
        if (isSecretPresent(ownerAddress, secretAddress)) {
```
Same applies for other parts

Reply:
Please see https://github.com/iExecBlockchainComputing/iexec-sms/pull/146/commits/5759ec2812187cbfbd5be50ab5a874b01802e210.